node/pkg/common/nodekey_test.go

```diff
@@ -0,0 +1,53 @@
+package common
+
+import (
+	"fmt"
+	"github.com/libp2p/go-libp2p-core/crypto"
+	"github.com/stretchr/testify/assert"
+	"go.uber.org/zap"
+	"io/ioutil"
+	"math/rand"
+	"os"
+	"testing"
+)
+
+func TestGetOrCreateNodeKeyWithNewPath(t *testing.T) {
+	// Get a non-existing temp file path to write auto-generated privKey to
+	path := "/tmp/node_key_test_" + fmt.Sprint(rand.Int())
```

Comment:
The problem I thought I ran into during this development was that CreateTemp was getting me, if ioutil.Tempfile() doesn't do a create this could be a compelling option, will look into it.